File: agileforge-backend/src/main/java/com/agileforge/application/service/AiAgentService.java
package com.agileforge.application.service;

import com.agileforge.domain.model.AiAgent;
import com.agileforge.domain.model.AiAgentTask;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

import java.time.Instant;
import java.util.*;

@Service
public class AiAgentService {

    private static final Logger log = LoggerFactory.getLogger(AiAgentService.class);

    public record AgentExecutionResult(UUID taskId, String status, String output, long executionTimeMs) {}

    public record SprintPlanSuggestion(List<UUID> suggestedTickets, int totalPoints, double completionProbability,
                                       String reasoning, List<String> risks) {}

    public record RiskReport(String overallRisk, List<RiskItem> items, List<String> recommendations) {}

    public record RiskItem(String title, String severity, String description, UUID relatedTicketId) {}

    private final AiAssistantService aiAssistantService;

    public AiAgentService(AiAssistantService aiAssistantService) {
        this.aiAssistantService = aiAssistantService;
    }

    public List<AiAgent> getAvailableAgents() {
        return List.of();
    }

    public AiAgent getAgentById(UUID agentId) {
        return null;
    }

    @Async("webhookExecutor")
    public void executeAgentTask(UUID agentId, UUID projectId, UUID organizationId, UUID triggeredBy,
                                 String taskType, Map<String, Object> input) {
        log.info("Executing AI agent task: agentId={}, taskType={}", agentId, taskType);
        Instant start = Instant.now();

        try {
            String result = processTask(agentId, taskType, input);
            long duration = Instant.now().toEpochMilli() - start.toEpochMilli();
            log.info("Agent task completed in {}ms", duration);
        } catch (Exception e) {
            log.error("Agent task failed: {}", e.getMessage(), e);
        }
    }

    public SprintPlanSuggestion suggestSprintPlan(UUID projectId, UUID sprintId) {
        return new SprintPlanSuggestion(
                List.of(), 0, 0.0, "Analysis pending", List.of()
        );
    }

    public RiskReport analyzeProjectRisks(UUID projectId) {
        return new RiskReport("UNKNOWN", List.of(), List.of("Run risk analysis agent"));
    }

    public String generateDocumentation(UUID projectId, String docType) {
        return "Documentation generation pending...";
    }

    public String generateRetrospective(UUID sprintId) {
        return "Retrospective generation pending...";
    }

    public List<AiAgentTask> getTaskHistory(UUID projectId, int page, int size) {
        return List.of();
    }

    public AiAgentTask getTaskResult(UUID taskId) {
        return null;
    }

    private String processTask(UUID agentId, String taskType, Map<String, Object> input) {
        return switch (taskType) {
            case "SPRINT_PLANNING" -> "Sprint plan generated";
            case "RISK_DETECTION" -> "Risk analysis completed";
            case "DOCUMENTATION" -> "Documentation generated";
            case "CODE_REVIEW" -> "Code review completed";
            case "RETROSPECTIVE" -> "Retrospective insights generated";
            default -> "Unknown task type";
        };
    }
}

File: agileforge-frontend/src/app/core/services/ai-agent.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AiAgent, AiAgentTask, SprintPlanSuggestion, RiskReport } from '../models/ecosystem.model';

@Injectable({ providedIn: 'root' })
export class AiAgentService {
  private http = inject(HttpClient);
  private baseUrl = '/api/ai-agents';

  getAvailableAgents(): Observable<AiAgent[]> {
    return this.http.get<AiAgent[]>(this.baseUrl);
  }

  getAgent(agentId: string): Observable<AiAgent> {
    return this.http.get<AiAgent>(`${this.baseUrl}/${agentId}`);
  }

  executeTask(agentId: string, organizationId: string, triggeredBy: string,
              taskType: string, projectId?: string, input?: Record<string, unknown>): Observable<void> {
    return this.http.post<void>(`${this.baseUrl}/${agentId}/execute`, {
      organizationId, triggeredBy, taskType, projectId, input
    });
  }

  suggestSprintPlan(projectId: string, sprintId?: string): Observable<SprintPlanSuggestion> {
    const params: Record<string, string> = {};
    if (sprintId) params['sprintId'] = sprintId;
    return this.http.get<SprintPlanSuggestion>(`${this.baseUrl}/sprint-plan/${projectId}`, { params });
  }

  analyzeRisks(projectId: string): Observable<RiskReport> {
    return this.http.get<RiskReport>(`${this.baseUrl}/risks/${projectId}`);
  }

  generateDocumentation(projectId: string, docType: string): Observable<{ content: string }> {
    return this.http.post<{ content: string }>(`${this.baseUrl}/documentation/${projectId}`, { docType });
  }

  generateRetrospective(sprintId: string): Observable<{ content: string }> {
    return this.http.get<{ content: string }>(`${this.baseUrl}/retrospective/${sprintId}`);
  }

  getTaskHistory(projectId: string, page: number = 0, size: number = 20): Observable<AiAgentTask[]> {
    return this.http.get<AiAgentTask[]>(`${this.baseUrl}/tasks/${projectId}`, { params: { page, size } });
  }

  getTaskResult(taskId: string): Observable<AiAgentTask> {
    return this.http.get<AiAgentTask>(`${this.baseUrl}/tasks/result/${taskId}`);
  }
}
